plan/optimizer.go

```diff
@@ -188,14 +182,17 @@ const (
 	CodeInvalidWildCard     terror.ErrCode = 3
 	CodeUnsupported         terror.ErrCode = 4
 	CodeInvalidGroupFuncUse terror.ErrCode = 5
-	CodeIllegalReference    terror.ErrCode = 6
 	CodeStmtNotFound        terror.ErrCode = 7
 	CodeWrongParamCount     terror.ErrCode = 8
 	CodeSchemaChanged       terror.ErrCode = 9
 
 	// MySQL error code.
+	CodeIllegalReference     terror.ErrCode = mysql.ErrIllegalReference
 	CodeNoDB                 terror.ErrCode = mysql.ErrNoDB
 	CodeUnknownExplainFormat terror.ErrCode = mysql.ErrUnknownExplainFormat
+	CodeWrongGroupField                     = mysql.ErrWrongGroupField
```

Comment:
how about add `terror.ErrCode` to be compatible with other error code definitions ?